### DISCUSSION TITLE
TPA: Entidad base y entidad juridica

### DISCUSSION THREAD

**Discussion by picardovidal** [dds-jv/foro#123] [TPA - 2020] [answered]
Comparto una consulta que hice por mail y la respuesta de los ayudantes por si alguien tiene una duda en el mismo tema.

En el punto 7, da un ejemplo donde dos organizaciones jurídicas se agrupan en una entidad base para atender a los requisitos legales. Más abajo en el punto 10 se indica que una entidad base puede pertenecer a sólo una entidad jurídica. Si seguimos el ejemplo dado en el punto 7 no debería una entidad base poder estar asociada a mas de una entidad jurídica? 
Por otro lado, podemos tener en el sistema una entidad Base sin que la misma tenga una entidad juridica asociada? ya que indica que todos los requerimientos pueden ser asociados a una entidad base o a una entidad juridica.

**Comment by picardovidal** [comment]
Esta es la respuesta de los ayudantes:

Leyendo el punto 7:
Cada organización podrá manejar una estructura de entidades, teniendo algunas entidades jurídicas y algunas entidades base. Las primeras son las entidades reconocidas formalmente e inscriptas en la Inspección General de Justicia (IGJ), las segundas pueden ser estructuras que en la gestión son organizaciones distintas, pero no lo son en la formalidad. Por ejemplo, si dos microempresarios realizan producción textil (uno, remeras; el otro, pantalones) y lo hacen de manera informal (en términos normativos), entonces pueden agruparse en una cooperativa única para poder atender los requerimientos legales, pero con continuidad de sus negocios de forma separada.

En este ejemplo está hablando de dos entidades base (microempresario de remeras y microempresario de pantalones) que se agrupan en una entidad jurídica (la cooperativa)
Siguiendo esto, las dos entidades base, solo necesitan estar asociadas a una entidad jurídica (en este caso la cooperativa).

Calculo que la confusión puede venir por la diferencia entre el termino organización y entidad. Dependerá de la implementación que hagan de esos dos conceptos cómo les quede la implementación de las restricciones, pero una interpretación valida puede ser, que sean directamente lo mismo.

Por otro lado: De lo que se lee en los puntos 7, 10 y 11, se puede interpretar que una entidad base no necesita estar en una entidad jurídica.

Un ejemplo de los datos de una organización/entidad puede ser:

Entidad: 
Hernandez e hijos (jurídica)
     Hernandez SRL (jurídica)
                   Hernandez Pinturas (base, dentro de la SRL)
                   Hernandez Chapas (base, dentro de la SRL)
     Taller Guadalupe (base)

otro ejemplo puede ser directamente:

Entidad:
Taller Pepita Fast and Furious (base)


Como el enunciado no menciona que las entidades base puedan asociar entidades jurídicas u otras entidades base, lo ideal sería no modelarlo en esta primer entrega. Pero tengan en cuenta que no es un requerimiento muy loco (x ej, tengo 2 entidades jurídicas, quiero agrupar sus contabilidades en una sola entidad base para tener todo mi patrimonio en un solo informe), así que está bueno que le den una vuelta de tuerca para que su diseño sea flexible en ese punto.

**Comment by roli93** [comment]
Thanks for sharing! (Lo cierro xq la duda está resuelta)

**Comment by BrianLuna** [comment]
> (x ej, tengo 2 entidades jurídicas, quiero agrupar sus contabilidades en una sola entidad base para tener todo mi patrimonio en un solo informe)

Por las dudas digo para aclarar, es justamente al revés, cierto?

2 entidades base van a agrupar sus contabilidades en una sola entidad jurídica...

Saludos
Brian

**Comment by roli93** [comment, marked as answer]
@BrianLuna Sí, está al revés. De todas formas, tengan en cuenta que este tipo de ambigüedades son para que las discutan con su ayudante y lleguen a una conclusión, por lo que no necesariamente todos van a terminar con el mismo modelo de relaciones entre entidades, no buscamos una solución estandarizada. 

PD: Recuerden si tienen dudas sobre un issue cerrado abrir un nuevo issue y referenciar al anterior (# + nro. de issue) y no comentar sobre el issue cerrado xq nadie lo ve (de hecho este mensaje lo vi muuuuy de pedo)
